### PR TITLE
Unique jobs djh

### PR DESCRIPTION
**Pull request by danhawkins**

### CODE REVIEW

File: lib/exq/enqueuer/server.ex [hunk 1]
```diff
@@ -65,6 +65,13 @@ defmodule Exq.Enqueuer.Server do
     {:reply, response, state}
   end
 
+  def handle_call({:enqueue_unique, queue, worker, args, uniquekey}, from, state) do
+    perform = fn -> handle_call({:enqueue, queue, worker, args}, from, state) end
+    response = with_unique_lock(perform, state, queue, worker, args, uniquekey)
+    {:reply, response, state}
+  end
+
+
```
Comment (brett-richardson) [inline lib/exq/enqueuer/server.ex]: 🔥

File: lib/exq/enqueuer/uniqueness.ex [hunk 2]
```diff
@@ -4,22 +4,43 @@ defmodule Exq.Enqueuer.Uniqueness do
   alias Exq.Redis.Connection, as: Redis
   alias Exp.Middleware.Pipeline
 
-  def with_unique_lock(callback, state, queue \\ "default", worker \\ "", args \\ []) do
+  # def with_unique_lock(callback, state, queue \\ "default", worker \\ "", args \\ []) do
+  #   with_unique_lock(callback, state, queue, worker, args, nil)
+  # end
```
Comment (brett-richardson) [inline lib/exq/enqueuer/uniqueness.ex]: 🔥

File: lib/exq/enqueuer/server.ex [hunk 3]
```diff
@@ -17,7 +17,7 @@ defmodule Exq.Enqueuer.Server do
 
   alias Exq.Support.Config
   alias Exq.Redis.JobQueue
-  import Exq.Enqueuer.Uniqueness, only: [with_unique_lock: 5]
+  import Exq.Enqueuer.Uniqueness, only: [with_unique_lock: 5, with_unique_lock: 6]
```
Comment (brett-richardson) [inline lib/exq/enqueuer/server.ex]: we should only import the 6 one I think?

File: lib/exq/enqueuer/uniqueness.ex [hunk 4]
```diff
@@ -4,22 +4,43 @@ defmodule Exq.Enqueuer.Uniqueness do
...
+  def with_unique_lock(callback, state, queue \\ "default", worker \\ "", args \\ [], uniquekey \\ nil) do
     redis = state.redis
-    args_key = Enum.join(args, ",")
-    key = combined_key(state.namespace, queue, Atom.to_string(worker), args_key)
+    simple_key = uniquekey || Enum.join(args, ",")
+    enqueue_key = combined_key(state.namespace, queue, Atom.to_string(worker), simple_key)
 
-    if unique_key_exists?(redis, key) do
-      {:job_not_unique, key}
+    if unique_key_exists?(redis, enqueue_key) do
```
Comment (brett-richardson) [inline lib/exq/enqueuer/uniqueness.ex]: probably want to rename this `enqueue_key_exists?`

**Comment by danhawkins** [comment]
@brett-richardson can we merge now?

**Pull request merged**